### PR TITLE
feat: support a pre-release component of the version string

### PR DESCRIPTION
**Pull request by devbww**
This allows `google::cloud::version_string()` to return, for example, "v2.5.0-rc" in pre-release builds.

Update the release guide to describe the additional steps required.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10181)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ set(PACKAGE_BUGREPORT "https://github.com/googleapis/google-cloud-cpp/issues")
 project(
     google-cloud-cpp
     VERSION 2.5.0
+    DESCRIPTION "rc"  # pre-release version
```
Comment (coryan) [inline CMakeLists.txt:25]: That is a bit cheeky.  `DESCRIPTION` should be used for something like "The Google Cloud C++ SDK".

There is a long discussion of a related topic here:

https://gitlab.kitware.com/cmake/cmake/-/issues/16716

It includes some alternatives that may work for us, e.g.:

```CMake
project(google-cloud-cpp VERSION 2.5.0 LANGUAGES CXX)
set(GOOGLE_CLOUD_CPP_PRE_RELEASE "-rc")
set(PROJECT_VERSION "${PROJECT_VERSION}${GOOGLE_CLOUD_CPP_PRE_RELEASE")
```

and then in `version_info.h.in`

```cc
#define GOOGLE_CLOUD_CPP_PRE_RELEASE "@GOOGLE_CLOUD_CPP_PRE_RELEASE@"
```

File: google/cloud/storage/version.h [hunk 2]
```diff
@@ -64,10 +62,19 @@ int constexpr version_minor() { return google::cloud::version_minor(); }
  */
 int constexpr version_patch() { return google::cloud::version_patch(); }
 
+/**
+ * Returns the Google Cloud Storage C++ Client pre-release version.
+ *
+ * @see https://semver.org/spec/v2.0.0.html for details.
+ */
+constexpr char const* version_pre_release() {
```
Comment (coryan) [inline google/cloud/storage/version.h:70]: Is that the `east const` form?

File: google/cloud/bigtable/version.h [hunk 3]
```diff
@@ -38,11 +38,9 @@
 
 namespace google {
 namespace cloud {
-/**
- * Contains all the Cloud Bigtable C++ client APIs.
- */
```
Comment (dbolduc) [inline google/cloud/bigtable/version.h:43]: I kind of like this comment. It shows up in the dox: https://screenshot.googleplex.com/3rin8yunN3qzskp

File: CMakeLists.txt (hunk 1 above)
Comment (devbww) [reply CMakeLists.txt]: Yes, a little cheeky.  Anyway, take a look at new impl along the lines suggested.

File: google/cloud/bigtable/version.h (hunk 3 above)
Comment (devbww) [reply google/cloud/bigtable/version.h]: Restored here, there, and everywhere.

File: google/cloud/storage/version.h (hunk 2 above)
Comment (devbww) [reply google/cloud/storage/version.h:73]: It's the "east const but constexpr west" form.  https://youtu.be/z6s6bacI424

File: CMakeLists.txt (hunk 1 above)
Comment (coryan) [reply CMakeLists.txt]: SGTM

**Review by coryan — APPROVED** [review]

**Pull request merged**